src/Interfaces/dictvectors.jl

```diff
@@ -74,3 +82,54 @@ conventional manner, but supports faster dot products.
 If `dv` is an [`MPIData`](@ref Main.Rimu.RMPI.MPIData), synchronize its contents among the ranks first.
 """
 freeze(v::AbstractVector) = copy(v)
+
+"""
+    working_memory(dv::AbstractDVec)
+
+Create a working memory instance compatible with `dv`. The working memory must be
+compatible with [`sort_into_targets!`](@ref) and [`fciqmc_step!`](@ref).
+"""
+working_memory(dv) = similar(localpart(dv))
+
+"""
+    fciqmc_step!(working_memory, curr_dv, prev_dv, ham, shift, dτ) -> stat_names, stats, working_memory, curr_dv, prev_dv
```

Comment:
```suggestion
    fciqmc_step!(v2, working_memory, v, ham, shift, dτ) -> curr_v, working_memory, prev_v, stat_names, stats
```
Here is a just a thought for the calling signature. Only the first two arguments are mutated. The `v` argument is not mutated. Currently a reference is returned as `prev_v`,  but this could  be  skipped (and assigning `prev_v` could be delegated to the calling code).  

The docstring would have to explain that `v^{(n)} ==  v` and  a reference to it is passed back as `prev_v`, that `v^{(n+1)} ==  curr_v`,  and that `w` and `working_memory` have to be suitable containers (with `v2` having the same type as `v`) that will be mutated  and references could change.

Reply:
Right, that makes sense. Perhaps in this signature, `curr_dv` and `prev_dv` could be called `source` and `target` as well.